### PR TITLE
Add the possibility to display the arrows in MeshcatVizualizer

### PR DESCRIPTION
**Pull request by GiulioRomualdi**
This PR allows you to easily draw arrows in MeshcatVizualizer

The original idea was taken from @rob-mau and generalized to avoid using non-standard datatype. 

For instance the following code

```python
from idyntree.visualize import MeshcatVisualizer
viz = MeshcatVisualizer()
viz.load_arrow(radius=0.01, shape_name="arrow", color=[255/255, 157/255, 68/255])
viz.set_arrow_transform(origin=[0, 0.5, 0], vector=[0.5, 0.33, 1], shape_name="arrow")
```

draws the following arrow
image

### CODE REVIEW

**Comment by rob-mau** [comment]
Hi Giulio, I was reading your implementation of the rotation matrix `R`. You can avoid to compute the variable `s`, since:

$$
\frac{1 - \cos \theta}{\sin^2 \theta} = \frac{1 - \cos \theta}{1 - \cos^2 \theta} = \frac{1}{1 + \cos \theta} = \frac{1}{1 + \hat{\boldsymbol{x}}^\top \hat{\boldsymbol{y}}}
$$

In this case the rotation matrix can be computed as:

$$
R(\hat{\boldsymbol{x}}, \\, \hat{\boldsymbol{y}}) \triangleq I + S(\hat{\boldsymbol{x}} \wedge \hat{\boldsymbol{y}}) + \frac{1}{1 + \hat{\boldsymbol{x}}^\top \hat{\boldsymbol{y}}} \\, S(\hat{\boldsymbol{x}} \wedge \hat{\boldsymbol{y}})^2
$$

With this formulation, there is no singularity in $\theta = 0$, so you can avoid the conditional statement:
`if s < 1e-6:
R = np.eye(3) 
else: ...
`

You can also avoid to compute the cross product $\hat{\boldsymbol{x}} \wedge \hat{\boldsymbol{y}}$, since:

$$
S(\hat{\boldsymbol{x}} \wedge \hat{\boldsymbol{y}}) = \hat{\boldsymbol{y}} \\, \hat{\boldsymbol{x}}^\top - \hat{\boldsymbol{x}} \\, \hat{\boldsymbol{y}}^\top
$$

but I don't know which computation is better.

**Comment by GiulioRomualdi** [comment]
Hi @rob-mau thank you for the hint. However, I think we still need an if indeed in case $\hat{\boldsymbol{x}}^\top \hat{\boldsymbol{y}} = -1$
the following is not defined (at least in the computer) 

$$
R(\hat{\boldsymbol{x}}, \\, \hat{\boldsymbol{y}}) \triangleq I + S(\hat{\boldsymbol{x}} \wedge \hat{\boldsymbol{y}}) + \frac{1}{1 + \hat{\boldsymbol{x}}^\top \hat{\boldsymbol{y}}} \\, S(\hat{\boldsymbol{x}} \wedge \hat{\boldsymbol{y}})^2
$$

File: bindings/python/visualize/meshcat_visualizer.py [hunk 1]
```diff
@@ -392,6 +402,40 @@ def set_primitive_geometry_transform(
...
+
+        if np.linalg.norm(vector) < 1e-6:
+            warnings.warn("The vector is too small to be visualized.", category=UserWarning, stacklevel=2)
+            return
+
+        # extract rotation matrix from a normalized vector
+        vector = vector / np.linalg.norm(vector)
+        dummy_vector = np.array([0, 0, 1])
+
+        # compute the rotation matrix bewteen the two vectors
```
Comment (S-Dafarra) [inline bindings/python/visualize/meshcat_visualizer.py:424]:
```suggestion
        # compute the rotation matrix between the two vectors
```

File: bindings/python/visualize/meshcat_visualizer.py [hunk 2]
```diff
@@ -392,6 +402,40 @@ def set_primitive_geometry_transform(
...
+            return
+
+        # compute the scaling matrix
+        S = np.diag([1, 1, np.linalg.norm(vector)])
+        transform = np.zeros((4, 4))
+        transform[0:3, 3] = np.array(origin) + np.array(vector) / 2
+        transform[3, 3] = 1
+
+        if np.linalg.norm(vector) < 1e-6:
+            warnings.warn("The vector is too small to be visualized.", category=UserWarning, stacklevel=2)
```
Comment (S-Dafarra) [inline bindings/python/visualize/meshcat_visualizer.py:417]: Is there something less than warning? It often happens that you don't care too much about the length of the arrow before displaying it. Having a warning continuously printed might be annoying

**Comment by S-Dafarra** [comment]
Btw, you can also compute the relative rotation by simply computing the corresponding axis and angle that align the two vectors. The first can be computed via the cross product, the second via the dot product. Since we are already loading iDynTree bindings, it may make sense to use directly the builtin function in iDynTree: https://github.com/robotology/idyntree/blob/4de2bf4d9bcd299b69a3f7ee52732331199d993b/src/core/include/iDynTree/Core/Rotation.h#L313

I did the same for the visualization of vectors in the iDynTree visualizer: https://github.com/robotology/idyntree/blob/4de2bf4d9bcd299b69a3f7ee52732331199d993b/src/visualization/src/VectorsVisualization.cpp#L40-L45

**Comment by rob-mau** [comment]
>  I think we still need an if indeed in case $\hat{\boldsymbol{x}}^\top \hat{\boldsymbol{y}} = -1$ the following is not defined

Yes, the singularity in $\theta = \pi$ is not remuvable. Also evaluating $R$ as Stefano suggested we have this problem. When the two vectors are aligned, the normal axis is not uniquely defined and the cross product between them is equal tothe zero vector. The Rodrigues' formula returns always the identity matrix, even if the two vectros are opposite.

File: bindings/python/visualize/meshcat_visualizer.py (hunk 2 above)
Comment (traversaro) [reply bindings/python/visualize/meshcat_visualizer.py:417]: I agree, if you display data from a sensor (i.e. force sensor) it makes total sense to have a near 0 value, there is any other reason to have this print/check?

**Review by S-Dafarra — APPROVED** [review]

**Comment by GiulioRomualdi** [comment]
yes

**Pull request merged**